.github/get_latest_index.py

```diff
@@ -0,0 +1,200 @@
+    ) -> None:
+        """
+        Updates SQL queries in the specified folder.
+
+        Args:
+            dir_path (str): The path to the folder containing SQL queries.
+            current_index_version (int): The current index version.
+            latest_idc_release_version (int): The latest IDC release version.
+        """
+        for file_name in os.listdir(dir_path):
+            if file_name.endswith(".sql"):
+                file_path = Path(dir_path) / file_name
+                with Path(file_path).open("r") as file:
+                    sql_query = file.read()
+                modified_sql_query = sql_query.replace(
+                    f"idc_v{current_index_version}",
+                    f"idc_v{latest_idc_release_version}",
+                )
+                with Path(file_path, "w").open() as file:
+                    file.write(modified_sql_query)
+                logger.debug("Updated SQL queries in file: %s", file_path)
+
+    def execute_sql_query(self, file_path: str) -> tuple[pd.DataFrame, str, str]:
+        """
+        Executes the SQL query in the specified file.
+
+        Args:
+            file_path (str): The path to the file containing the SQL query.
+
+        Returns:
+            Tuple[pd.DataFrame, str, str]: A tuple containing the DataFrame with query results,
+            the CSV file name, and the Parquet file name.
+        """
+        with Path(file_path).open("r") as file:
+            sql_query = file.read()
+        index_df = self.client.query(sql_query).to_dataframe()
+        file_name = Path(file_path).name.split(".")[0]
+        csv_file_name = f"{file_name}.csv.zip"
+        parquet_file_name = f"{file_name}.parquet"
+        logger.debug("Executed SQL query from file: %s", file_path)
+        return index_df, csv_file_name, parquet_file_name
+
+    def create_csv_zip_from_df(
+        self, index_df: pd.DataFrame, csv_file_name: str
+    ) -> None:
+        """
+        Creates a compressed CSV file from a pandas DataFrame.
+
+        Args:
+            index_df (pd.DataFrame): The pandas DataFrame to be saved as a CSV.
+            csv_file_name (str): The desired name for the resulting ZIP file (including the ".csv.zip" extension).
+        """
+        index_df.to_csv(csv_file_name, compression={"method": "zip"}, escapechar="\\")
+        logger.debug("Created CSV zip file: %s", csv_file_name)
+
+    def create_parquet_from_df(
+        self, index_df: pd.DataFrame, parquet_file_name: str
+    ) -> None:
+        """
+        Creates a Parquet file from a pandas DataFrame.
+
+        Args:
+            index_df (pd.DataFrame): The pandas DataFrame to be saved as a Parquet file.
+            parquet_file_name (str): The desired name for the resulting Parquet file.
+        """
+        index_df.to_parquet(parquet_file_name)
+        logger.debug("Created Parquet file: %s", parquet_file_name)
+
+    def run_queries_folder(self, dir_path: str) -> None:
```

Comment:
rename to `generate_index_data_files` and support generating only csv archive, only parquet or both

Reply:
generating both for now as we could selectively package at cmake step